### ISSUE TITLE
restarted simulation settings compatibility with original

### ISSUE THREAD

**Issue by PhilipDeegan** [PHAREHUB/PHARE#666]
it's expected that some parameters would be considered invalid if they are changed between the original and the restarted simulation.
for instance, the L0 domain cells and dl

the code will reference this issue where this is to be maintained and considered.

**Comment by PhilipDeegan** [comment]
:imp:

**Comment by nicolasaunai** [comment]
👹

**Comment by PhilipDeegan** [comment]
we have the serialized simulation in each diags file, so we should have some `is_enough_equal` ~~ rather than == to say if they are interoperable

**Comment by nicolasaunai** [comment]
yeah some `is_restart_compatible()` that would assert key params are or not ok.

**Comment by PhilipDeegan** [comment]
this is likely to evolve on an as needed basis
